drivers/secure_os/ws_api.cr

```diff
@@ -58,7 +58,9 @@ class SecureOS::WsApi < PlaceOS::Driver
     schedule.clear
   end
 
-  def subscribe_states(states : Array(String) = ["attached", "armed", "alarmed"])
+  def subscribe_states(
```

Comment:
This method should be protected as it shouldn't be called from another module
Also worth noting that drivers don't allow overloading of public function names - as you can't differentiate them when making remote procedure calls

Reply:
with auto subscribing to everything I guess we probably don't need any of the subscribe methods exposed

Reply:
works for me

Reply:
We can replace all the subscription methods with a single subscribe method that takes the states/events to subscribe to. And then use a setting to allow changing the states/events or leave the settings out in case it's not needed.

```crystal
private def subscribe_all(
    states : Array(StateType)? = [StateType::Attached, StateType::Armed, StateType::Alarmed],
    events : Array(String)? = nil
  )
    rules = [] of NamedTuple(
      type: String,
      id: String,
      states: Array(StateType)?,
      events: Array(String)?,
      action: Symbol,
      )

    @camera_list.each do |camera|
      rules << {
        type: camera.type,
        id: camera.id,
        states: states,
        events: nil,
        action: :STATE_CHANGED,
      }
      rules << {
        type: camera.type,
        id: camera.id,
        states: nil,
        events: events,
        action: :EVENT,
      }
    end

    return if rules.empty?

    send({
      type: :subscribe,
      # id: 1234, # optional id used in error responses
      data: {
        add_rules: rules,
      },
    }.to_json, wait: false)
  end
```